### DISCUSSION TITLE
How to add pagination to the top of the tables

### DISCUSSION THREAD

**Discussion by jcc5018** [filamentphp/filament#11353] [Help]
### Package

Table builder

### Package Version

v3

### How can we help you?

Since clicking back from a create screen doesnt seem to remember the last page you were on, nor sorting options.  Is there a way to 1. allow the system to do so, so that you dont have to keep navigating to find the next file you are editing. 

2. Add pagination to the top of the table to quickly navigate to the desired page without having to scroll long tables.

I searched the docs and only found methods to disable pagination, but i have no idea how to add it to the top and bottom of a table.  Thanks

**Comment by dissto** [comment]
Perhaps [query string](https://filamentphp.com/docs/3.x/tables/advanced#query-string) is what you needed?

**Reply by jcc5018** [reply]
Not sure how that adds the pagination buttons to the top of the table, might help with remembering things though

**Reply by ArtMin96** [reply]
You can try the following steps.
1. Extent Filament's `Filament\Tables\Table`
2. Change `$view` to 'your-view'
3. Copy everything from [table view file](https://github.com/filamentphp/filament/blob/3.x/packages/tables/resources/views/index.blade.php)
4. Move pagination to the top of the table: [#L1225](https://github.com/filamentphp/filament/blob/5bc6e609a8ac108d8ea723b3d68f611f00922b4c/packages/tables/resources/views/index.blade.php#L1225)

**Comment by jjanampa** [comment]
I did this:

- In the resources

```
  public static function table(Table $table): Table
    {
        return $table
            ->header(view('tables.header', [
                'table' => $table
            ]))
            ->paginated([10, 20, 50, 100])
            ...
    }
```

- In the view (resources/views/tables/header.blade.php):
```
<div class="table-header">
    @php
    $records = $table->getRecords()
    @endphp
    {{-- Paginador en la cabecera usando el componente de Filament --}}
    @if ($records instanceof \Illuminate\Contracts\Pagination\Paginator)
        <x-filament::pagination
            :extreme-links="true"
            :page-options="[10, 20, 50, 100]"
            :paginator="$records"
            class="fi-ta-pagination px-3 py-3 sm:px-6"
        />
    @endif
</div>
```
